### PR TITLE
Migrate Redshift API to ASF

### PR DESCRIPTION
**Pull request by whummer**
Migrate Redshift API to ASF, using moto fallback dispatcher.

### CODE REVIEW

File: localstack/services/redshift/provider.py [hunk 1]
```diff
@@ -21,15 +23,11 @@ def fix_keys(o, **kwargs):
...
-    redshift_responses.itemize = itemize
 
-
-def start_redshift(port=None, asynchronous=False):
-    port = port or config.service_port("redshift")
-    apply_patches()
-    return start_moto_server("redshift", port, name="Redshift", asynchronous=asynchronous)
+class RedshiftProvider(RedshiftApi, ServiceLifecycleHook):
+    def on_after_init(self):
+        apply_patches()
```
Comment (thrau) [inline localstack/services/redshift/provider.py:33]: it seems to me we could just remove apply_patches all together, and define the `@patch` decorated methods on module level. lazy-init will make sure the patches are applied only when the service is loaded.

Comment (whummer) [reply localstack/services/redshift/provider.py]: Good catch, updated. 👍 

I guess over time we'll see whether we want to introduce a patch apply/unapply in the service lifecycle, but for now there doesn't seem to be a strong use case. Also makes the implementation here a bit simpler, as we don't need to rely on the `ServiceLifecycleHook`.

Comment (thrau) [reply localstack/services/redshift/provider.py]: the `@patch` actually adds the `Patch` object to the function, so you could do something like
```python
def toggle_patches():
    itemize.patch.undo()
```

another way can be to use the `Patches` collection to aggregate the used patches. `patches` could become a conventional object as part of the module

```python
patches = Patches([itemize.patch, ...])
```
then you can call `module.patches.apply()` or `module.patches.undo()`

lots of flexibility :-)

**Comment by whummer** [comment]
Tests against Pro currently failing, as we don't yet have fallbacks in place to gracefully handle errors in PG server creation. (added in corresponding PR in -ext - tests should be fixed after next ext release).

**Review by thrau — APPROVED** [review]
LGTM. i spent about 15 minutes trying to understand what the itemize patch does exactly, but can only guess it fixes some serialization problem in moto for that one particular case. also there was no test for it so i couldn't figure out whether it's still needed. anyway, guess we'll leave it in there for now to avoid regression. moving forward i think we should be a bit less lenient about leaving patches to parsing/serialization mechanisms lying around that don't have tests associated with them.

**Pull request merged**